src/CppUTest/MemoryLeakDetector.cpp

```diff
@@ -66,7 +68,7 @@ void SimpleStringBuffer::addMemoryDump(const void* memory, size_t memorySize)
 	size_t p;
 
     while (currentPos < memorySize) {
-        add("    %04lx: ", currentPos);
+        add("    %04lx: ", MAYBE_16BIT(currentPos));
```

Comment:
As it turns out, it didn't even work :-( If the `lx` stays, then we need to pass it an unsigned long with the high word masked away, or it will display random data in the high word.